tests/unit/amo/components/TestCategory.js

```diff
@@ -235,22 +238,23 @@ describe(__filename, () => {
     sinon.assert.notCalled(fakeDispatch);
   });
 
-  it('dispatches getLanding when results are not loaded', () => {
+  it('dispatches getLanding and sets a viewContext when results are not loaded', () => {
     _categoriesFetch();
     _categoriesLoad();
 
     const fakeDispatch = sinon.stub(store, 'dispatch');
     render({}, { autoDispatchCategories: false });
 
-    sinon.assert.callCount(fakeDispatch, 1);
+    sinon.assert.callCount(fakeDispatch, 2);
+    sinon.assert.calledWith(fakeDispatch, setViewContext(fakeCategory.type));
```

Comment:
This test exercises [a specific if branch](https://github.com/bobsilverberg/addons-frontend/blob/d41b5e99ba24219eab50f1a1127f937497195fa6/src/amo/components/Category/index.js#L110).

It's not necessary to check the view context logic because that's not affected by the if branch. In fact, by checking the view context behavior, you are duplicating the [other test](https://github.com/bobsilverberg/addons-frontend/blob/d41b5e99ba24219eab50f1a1127f937497195fa6/tests/unit/amo/components/TestCategory.js#L141). Such a duplication won't really hurt anything but it could potentially introduce a maintenance burden.

I would suggest these changes:
* Remove the `assert.callCount()` from this test since it's not necessary
* Remove the assertion of the `setViewContext()` call